shared/rocroller/lib/include/rocRoller/DataTypes/DataTypes_Buffer.hpp

```diff
@@ -0,0 +1,78 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright 2019-2025 AMD ROCm(TM) Software
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#pragma once
+
+#include <iostream>
+
+#ifdef ROCROLLER_USE_HIP
+#include <hip/hip_runtime.h>
+#endif
+
+namespace rocRoller
+{
+    struct Buffer
+    {
+        uint32_t desc0;
+        uint32_t desc1;
+        uint32_t desc2;
+        uint32_t desc3;
+
+        Buffer()
+            : desc0(0)
+            , desc1(0)
+            , desc2(0)
+            , desc3(0)
+        {
+        }
+
+        Buffer(uint32_t d0, uint32_t d1, uint32_t d2, uint32_t d3)
+            : desc0(d0)
+            , desc1(d1)
+            , desc2(d2)
+            , desc3(d3)
+        {
+        }
```

Comment:
```suggestion
        uint32_t desc0 = 0;
        uint32_t desc1 = 0;
        uint32_t desc2 = 0;
        uint32_t desc3 = 0;
        
        inline auto operator<=>(Buffer const& other) const = default;
```

Let's use language-generated functions where possible!

Reply:
Updated in 0cc8b31